## MongoDB Python 활용
- pymongo 설치


In [2]:
!pip install pymongo

Defaulting to user installation because normal site-packages is not writeable
  Using cached pymongo-4.11.3-cp312-cp312-win_amd64.whl.metadata (22 kB)
  Using cached dnspython-2.7.0-py3-none-any.whl.metadata (5.8 kB)
Using cached pymongo-4.11.3-cp312-cp312-win_amd64.whl (882 kB)
Using cached dnspython-2.7.0-py3-none-any.whl (313 kB)


In [ ]:
from pymongo import MongoClient

In [43]:
uri = "mongodb://localhost:27017/"
client = MongoClient(
    host='127.0.0.1',
    port=27017
)
db = client.get_database("ubion")
coll_list = db.list_collection_names()
coll_one = db.user.find_one()
print(coll_one)

{'_id': ObjectId('67ef37bd6a9e9e446c7efeb7'), 'name': 'test', 'password': '1234'}


In [19]:
collections = db['users']

In [20]:
data = [
    {
        "name" : "kim",
        "phone" : "01000000000",
        "gender" : "male"
    },
    {
        "name" : "seo",
        "phone" : "01000000000",
        "gender" : "female"
    }
]
collections.insert_many(data)

InsertManyResult([ObjectId('67ef43f96427c5b1ba44073e'), ObjectId('67ef43f96427c5b1ba44073f')], acknowledged=True)

In [27]:
list(
    collections.find(
        {},
        {'_id' : 0}
    )
)

[{'name': 'kim', 'phone': '01000000000', 'gender': 'male'},
 {'name': 'seo', 'phone': '01000000000', 'gender': 'female'}]

In [28]:
collections.delete_one({'name' : "seo"})

DeleteResult({'n': 1, 'ok': 1.0}, acknowledged=True)

In [29]:
list(
    collections.find(
        {},
        {'_id' : 0}
    )
)

[{'name': 'kim', 'phone': '01000000000', 'gender': 'male'}]

In [31]:
list(db.user.find())

[{'_id': ObjectId('67ef37bd6a9e9e446c7efeb7'),
  'name': 'test',
  'password': '1234'},
 {'_id': ObjectId('67ef37bd6a9e9e446c7efeb8'),
  'name': 'test2',
  'password': '1111'}]

### mongoDB select 조건식
-find(옵션1, 옵션2, ...)
1. db.collection.find() -> SELECT * FROM TABLE
2. db.collection.find({'id' : 'test', 'pass' : '1234'}) -> SELECT * FROM TABLE WHERE id = 'test' AND pass = '1234'
3. db.collection.find({'id' : 'test}, {'pass' : '1234'}) -> SELECT * FROM TABLE WHERE id = 'test' OR pass = '1234'
4. db.collection.find({}, {'id' : 1 , 'pass' : 1}) -> SELECT id, pass FROM TABLE
5. db.collection.find({}, {'id' : 0}) -> SELECT
6. db.collection.find({'age' : {'$gt' : 25}}) -> SELECT * FROM TABLE WHERE age > 25
7. db.collection.find({'age' : {'$lt' : 25}}) -> SELECT * FROM TABLE WHERE age < 25
8. db.collection.find({'id' : {/bc/}}) -> SELECT * FROM WHERE id like '%bc%'
9. db.collection.find().sort({'id' : 1}) -> SELECT * FROM TABLE ORDER BY id
10. db.collection.find().sort({'id' : -1}) -> SELECT * FROM TABLE ORDER BY id DESC
11. db.collection.find().limit(10) -> SELECT * FROM TABLE LIMIT 10

In [75]:
# mongodb에서 admin 데이터베이스에서 test라는 collection에 있는 데이터 중
# SAL 값이 1500 보다 큰 데이터에서 'ENAME' 을 기준으로 내림차순 정렬
db = client['admin']
collection = db['test']

print(list(collection.find({})))
print(list(collection.find({'SAL': {'$gt': 1500}}).sort([('ENAME', -1), ('SAL', 1)])))

[{'_id': ObjectId('67ef60da541ca7d2b80d266b'), 'ENAME': 'kim', 'SAL': 2000}, {'_id': ObjectId('67ef6104541ca7d2b80d266c'), 'ENAME': 'seo', 'SAL': 2500}, {'_id': ObjectId('67ef615e541ca7d2b80d266d'), 'ENAME': 'pak', 'SAL': 1500}]
[{'_id': ObjectId('67ef6104541ca7d2b80d266c'), 'ENAME': 'seo', 'SAL': 2500}, {'_id': ObjectId('67ef60da541ca7d2b80d266b'), 'ENAME': 'kim', 'SAL': 2000}]
